dns-controller/pkg/dns/dnscontroller.go

```diff
@@ -423,6 +428,30 @@ func (o *dnsOp) deleteRecords(k recordKey) error {
 		return fmt.Errorf("no suitable zone found for %q", fqdn)
 	}
 
+	if dnsProviderId == k8scoredns.ProviderName {
+		// TODO: work-around before ResourceRecordSets.List() is implemented for CoreDNS
```

Comment:
Nit: moving this comment on top of if-block will make it clearer that the whole if-block is part of the work-around.